coderd/autobuild/lifecycle_executor.go

```diff
@@ -195,9 +212,13 @@ func (e *Executor) runOnce(t time.Time) Stats {
 							useActiveVersion(accessControl, ws) {
 							log.Debug(e.ctx, "autostarting with active version")
 							builder = builder.ActiveVersion()
+
+							if latestBuild.TemplateVersionID != template.ActiveVersionID {
+								shouldAutoUpdate = true // control flag for notifications
```

Comment:
I think this flag could be more clearly named; as it is right now it has no real conceptual link to notifications, so perhaps something like `didAutoUpdate` or something would be more clear? We could also add a comment here to explain why we're capturing this state (so that we can know when a workspace was auto-updated and notify accordingly).

Reply:
Fixed, switched to `didAutoUpdate` with a comment